vignettes/Berekeningen.Rmd

```diff
@@ -0,0 +1,305 @@
+
+Er moet wel telkens aangegeven worden over welk type variabele het gaat (`TypeVariabele`, bv. Percentage, Decimaal getal, Categorie) en bij een categorische variabele moet ook de gebruikte schaal opgegeven worden (`Invoertype`, bv. TANSLEY IHD).
+
+Voor de berekeningen worden deze waarden intern omgezet naar intervallen.  De onder- en bovengrenzen die hierbij voor de verschillende schalen gehanteerd worden, kunnen opgevraagd worden via `geefVertaallijst(ConnectiePool)`.  De berekeningen gebeuren voor de boven- en ondergrens afzonderlijk, met als resultaat een nieuw interval.<!--nog aanpassen als we fuzzy logic gebruiken!-->  Voor de weergave van het resultaat worden voor rechtstreeks ingevoerde voorwaarden de door de gebruiker opgegeven waarden overgenomen; berekende waarden worden weergegeven in dezelfde schaal als de voorwaarde (LSVI-criterium).
+
+#Status
+
+Om de status van een voorwaarde te berekenen, worden de onder- en bovengrenzen met elkaar vergeleken (gebruik makend van functie `berekenStatus()`):
+
+- voor operator '<' de bovengrens van de waarde met de ondergrens van de grenswaarde
+- voor operator '<=' de bovengrens van de waarde met de bovengrens van de grenswaarde
+- voor operator '>' de ondergrens van de waarde met de bovengrens van de grenswaarde
+- voor operator '>=' de ondergrens van de waarde met de ondergrens van de grenswaarde
+- voor operator '=' beide ondergrenzen en beide bovengrenzen (idee is dat het interval van waarde gelijk is aan het interval van de grenswaarde of er volledig binnen valt)
+
+Het resultaat is telkens een logische waarde die aangeeft of de voorwaarde gunstig (TRUE) of ongunstig (FALSE) scoort.
+
+Om de status van een indicator te berekenen, wordt de aggregatie uitgevoerd die bij `geefInvoervereisten()` in het veld `Combinatie` weergegeven is (meestal een combinatie van voorwaarden met operatoren AND en OR), waarbij de resultaten TRUE/FALSE ingevuld worden bij de respectievelijke voorwaarden (gebruik makend van functie `combinerenVoorwaarden()`).
+
+Voor de aggregatie naar hogere niveaus (criterium en volledige opname) worden deze resultaten normaal verder geaggregeerd met een AND-operator (als parameter aggregatiemethode de default '1-out-all-out' is).  Als een gebruiker kiest voor de aggregatiemethode 'RapportageHR', dan wordt rekening gehouden met het feit of een indicator belangrijk (`Belang` = 'b') of zeer belangrijk is (`Belang` = 'zb').  De beoordeling is gunstig (TRUE) als meer dan 50 procent van de indicatoren gunstig zijn EN als geen enkele zeer belangrijke indicator ongunstig is.
+
+#Distance-to-target
+
+De distance-to-target of verschilscore van een voorwaarde wordt als volgt berekend met de functie `berekenVerschilscores()`:
+
+- theoretisch maximum bepalen voor de voorwaarde:<!--zou idealiter in de databank toegevoegd worden-->
+    - voor bedekkingen en aandelen is dit 1 of 100 %
+    - voor het aantal soorten/kenmerken is dit het aantal soorten/kenmerken van de lijst in de databank
```

Comment:
in de branch `oppervlakteafhankelijkheid_aantalsleutelsoorten`, hebben we dit vervangen door 2 * de grenswaarde en indien dit zou resulteren in een verschilscore > 1, wordt de verschilscore gelijkgesteld aan 1. Dit is een betere manier van berekenen dan de volledige lijst van soorten te nemen als theoretisch maximum omdat dit maximum nooit kan bereikt worden in een perceel.

Reply:
Op zich geen slecht idee voor lange lijsten sleutelsoorten, maar pas je ditzelfde ook toe als het gaat over bv. het aantal ouderdomsstadia of aantal groeiklassen?  Want in dit geval ga je vaak niet eens aan een verschilscore van 1 kunnen geraken.  Als je het enkel zou toepassen als het  aantal soorten/kenmerken van de lijst > 2 keer de grenswaarde, dan zou ik het wel overwegen om ook in de gewone rekenmodule toe te passen.

Reply:
Idee is inderdaad om dit enkel toe te passen voor aantal sleutelsoorten

Reply:
En waarom zoiets doen voor aantal en niet voor bedekking?  Bij een bedekking van sleutelsoorten gaat ook nooit het theoretisch maximum van 100 % bereikt worden?
Ik vind het een goed idee, en dit zou in elk geval ook een oplossing zijn voor voorwaarden waar we tot nu toe geen theoretisch maximum op geplakt hebben (bv. grootte vegetatievlek).  Maar dit in deze PR oplossen, zou ons wat ver leiden van het oorspronkelijke doel, vrees ik.  Ik zou voorstellen om het idee nog even voort te laten uitkristalliseren en dan de invulling voor het theoretisch maximum voor alle voorwaarden eens grondig te bekijken samen met issue #64.